ts/src/bitmart.ts

```diff
@@ -1184,21 +1201,22 @@ export default class bitmart extends Exchange {
         //          "legal_coin_price":"0.1302699"
         //      }
         //
-        let timestamp = this.safeInteger (ticker, 'timestamp');
+        let timestamp = this.safeIntegerN (ticker, [ 'timestamp', 'ts', 12 ]);
```

Comment:
Hum, this is tricky. The safeMethods won't work properly if the ticker can either be a list or a direct.

Reply:
I could separate the list values into a dictionary in `fetchTicker` and `fetchTickers` then parse the new dictionary if that works

Reply:
@Dan-krm you don't need to convert the list, you can do something like

```
let high = this.safeString(ticker, 'h') // read from dict
if (high == undefined) {
   high = this.safeString(ticker, 0) // read from list
}
```
besides that parseTicker/parseTickers expect a dict and not a list, so we might need to create a wrapper around the list, please check how we do it in bitfinex2

Reply:
Okay I'm working on it

Reply:
I created a dict wrapper for spot fetchTickers and separated reading the values from a dict or list in parseTicker